chromium_src/chrome/browser/printing/print_preview_message_handler.cc

```diff
@@ -113,17 +124,34 @@ bool PrintPreviewMessageHandler::OnMessageReceived(
   return handled;
 }
 
+void PrintPreviewMessageHandler::OnRequestPrintPreview(
+    content::RenderFrameHost* render_frame_host,
+    const PrintHostMsg_RequestPrintPreview_Params& params) {
+  if (params.webnode_only) {
+    PrintViewManager::FromWebContents(web_contents())->PrintPreviewForWebNode(
+        render_frame_host);
+  }
+}
+
 void PrintPreviewMessageHandler::PrintToPDF(
     const base::DictionaryValue& options,
     const atom::api::WebContents::PrintToPDFCallback& callback) {
-  int request_id;
+  PrintViewManager::FromWebContents(web_contents())->PrintPreviewNow(
+        web_contents()->GetMainFrame(), false);
+
+  int request_id = -1;
   options.GetInteger(kPreviewRequestID, &request_id);
```

Comment:
we should put `CHECK_GT(request_id, -1)` in case we don't get anything from options.